lib/vehicle_factory.rb

```diff
@@ -0,0 +1,33 @@
+class VehicleFactory
+  attr_reader :factory_vehicles,
+              :vehicle_data,
+              :api_data
+
+  def initialize(engine = :ev)
+    @factory_vehicles = []
+    @api_data = DmvDataService.new.wa_ev_registrations
+    @vehicle_data = {
+      vin: nil,
+      make: nil,
+      model: nil,
+      year: nil,
+      engine: engine
+      }
+  end
+
+  def create_vehicles(api_data)
+    api_data.each do |data|
+      vehicle = Vehicle.new(
+        vin: data[:vin_1_10],
+        make: data[:make],
+        model: data[:model],
+        year: data[:model_year],
+        engine: @vehicle_data[:engine]
+      )
+      @factory_vehicles << vehicle
+    end
+  end
+  @factory_vehicles
```

Comment:
I wonder if this would be better as a map, though I suppose if there were already elements in `@factory_vehicles`, using a map here wouldn't return those as well as all of the new ones. No strong opinion or suggestion here, just something to think about.